router/router.go

```diff
@@ -0,0 +1,71 @@
+package router
+
+import (
+	"github.com/charmbracelet/bubbles/key"
+	tea "github.com/charmbracelet/bubbletea"
+)
+
+// Screen represents a model with a particular binding to trigger it
+type Screen struct {
+	model   tea.Model
+	binding key.Binding
+}
+
+// Model stores different screens and the currently focused screens
+type Model struct {
+	screens []Screen
+	current int
+}
+
+// New creates a new empty Model
+func New() Model {
+	return Model{}
+}
+
+// NewWithScreens creates a new model with an array of Screen
+func NewWithScreens(screens []Screen) Model {
+	current := 0
+	return Model{
+		screens,
+		current,
+	}
+}
+
+func (m Model) updateCurrent(msg tea.Msg) (Model, tea.Cmd) {
+	if len(m.screens) <= m.current {
+		return m, nil
+	}
+	var cmd tea.Cmd
+	m.screens[m.current].model, cmd = m.screens[m.current].model.Update(msg)
+	return m, cmd
+}
+
+// AddScreen adds a new screen to the router
+func (m *Model) AddScreen(model tea.Model, binding key.Binding) {
+	m.screens = append(m.screens, Screen{model, binding})
+}
+
+// Init implements tea.Model
+func (Model) Init() tea.Cmd {
+	return nil
```

Comment:
I wonder if we should Init all the screens here?

Reply:
Yeah, makes sense, will do!

Reply:
I believe you're right, It probably makes sense to `Init` the current screen, we should probably keep track of whether the screen has been initialized and do that on first load (we should check this every time the navigator switches screens).

Reply:
Makes sense, so it'll be kind of lazily executed `Init`. If there's a need for something to be executed right away in the future, we can extend the api to accomodate